File: backend/services/competency-service/src/main/java/com/lmspilot/competency/config/DevelopmentSeed.java
package com.lmspilot.competency.config;

import com.lmspilot.competency.domain.*;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.*;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;
import java.util.List;

@Component
public class DevelopmentSeed implements ApplicationRunner {
    private final CompetencyRepository competencies;
    private final CompetencyProfileRepository profiles;
    private final CompetencyProfileRequirementRepository requirements;
    private final boolean enabled;

    public DevelopmentSeed(CompetencyRepository c, CompetencyProfileRepository p, CompetencyProfileRequirementRepository r,
                           @Value("${lmspilot.seed-demo:true}") boolean e) {
        competencies = c;
        profiles = p;
        requirements = r;
        enabled = e;
    }

    @Override
    @Transactional
    public void run(ApplicationArguments args) {
        if (!enabled || competencies.count() > 0) return;
        var d = competencies.save(new CompetencyEntity("DIGITAL-LITERACY", "Năng lực số", "Sử dụng công cụ số an toàn và hiệu quả", "Năng lực cốt lõi", 5, CompetencyStatus.ACTIVE));
        var s = competencies.save(new CompetencyEntity("INFOSEC-AWARENESS", "Nhận thức an toàn thông tin", "Nhận diện và xử lý rủi ro an toàn thông tin", "Tuân thủ", 5, CompetencyStatus.ACTIVE));
        var p = profiles.save(new CompetencyProfileEntity("ALL-EMPLOYEES", "Khung năng lực nhân viên", "Mức năng lực tối thiểu dùng cho dữ liệu mẫu", null, null, true));
        requirements.saveAll(List.of(new CompetencyProfileRequirementEntity(p, d, 2, 1), new CompetencyProfileRequirementEntity(p, s, 2, 1)));
    }
}

File: backend/services/competency-service/src/main/kotlin/com/lmspilot/competency/config/DevelopmentSeed.kt
package com.lmspilot.competency.config

import com.lmspilot.competency.domain.*
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Component
class DevelopmentSeed(
    private val competencies: CompetencyRepository,
    private val profiles: CompetencyProfileRepository,
    private val requirements: CompetencyProfileRequirementRepository,
    @Value("\${lmspilot.seed-demo:false}") private val enabled: Boolean,
) : ApplicationRunner {
    @Transactional
    override fun run(args: ApplicationArguments) {
        if (!enabled || competencies.count() > 0) return
        val digital = competencies.save(CompetencyEntity(code = "DIGITAL-LITERACY", name = "Năng lực số", category = "Năng lực cốt lõi", description = "Sử dụng công cụ số an toàn và hiệu quả"))
        val security = competencies.save(CompetencyEntity(code = "INFOSEC-AWARENESS", name = "Nhận thức an toàn thông tin", category = "Tuân thủ", description = "Nhận diện và xử lý rủi ro an toàn thông tin"))
        val profile = profiles.save(CompetencyProfileEntity(code = "ALL-EMPLOYEES", name = "Khung năng lực nhân viên", description = "Mức năng lực tối thiểu dùng cho dữ liệu mẫu"))
        requirements.saveAll(listOf(CompetencyProfileRequirementEntity(profile = profile, competency = digital, requiredLevel = 2), CompetencyProfileRequirementEntity(profile = profile, competency = security, requiredLevel = 2)))
    }
}
